src/app/services/user.service.ts

```diff
@@ -2,21 +2,106 @@ import { Injectable } from '@angular/core';
 
 import { BehaviorSubject, Observable } from 'rxjs';
 
-import { User } from '../models';
+import { User, UserGroup } from '../models';
+import { Router } from '@angular/router';
+import { HttpClient } from '@angular/common/http';
+import { EnvironmentService } from './environment.service';
+import { BlockSetId } from './block.service';
+
+export const GUEST_USER = 'guest';
```

Comment:
it could be nice if the frontend and backend shared a single source of truth on this name, but no need to do anything about it now

Reply:
or does the backend even have a concept of a guest user? if not, you can ignore my comment above!